### PR TITLE
APP-1179/APP-1180 : IB APIs to search for Jira users / assign new ticket to user

### PR DESCRIPTION
**Pull request by Hamitay-Daitan**

### CODE REVIEW

File: integration-authorization/pom.xml [hunk 1]
```diff
@@ -25,6 +25,12 @@
         </dependency>
 
         <dependency>
+            <groupId>com.google.http-client</groupId>
+            <artifactId>google-http-client-jackson</artifactId>
+            <version>1.15.0-rc</version>
+        </dependency>
```
Comment (robson-sanchez) [inline integration-authorization/pom.xml]: Copy this dependency to <dependencyManagement> section on the parent pom and update the version to 1.22.0

Example:
 ```
<dependency>
    <groupId>com.google.http-client</groupId>
    <artifactId>google-http-client-jackson2</artifactId>
    <version>1.22.0</version>
</dependency>
```

In this pom file, you should change the artifactId to 'google-http-client-jackson2' and remove the version.

File: integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java [hunk 2]
```diff
@@ -40,4 +44,24 @@
    * @throws AuthorizationException when a problem occur with this operation.
    */
   void authorize(AuthorizationPayload authorizationPayload) throws AuthorizationException;
+
+  /**
+   * Get an access token for user to perform calls to a resource provided by a given url.
```
Comment (robson-sanchez) [inline integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java]: 'Get an access token for user to perform calls to an external system resource'

File: integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java [hunk 3]
```diff
@@ -40,4 +44,24 @@
...
+   * @param url Resource url.
```
Comment (robson-sanchez) [inline integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java]: Integration URL

File: integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java [hunk 4]
```diff
@@ -40,4 +44,24 @@
...
+   * @param userId User id.
+   * @return An access token.
+   * @throws AuthorizationException Thrown in any case of error.
+   */
+  String getAccessToken(String url, Long userId) throws AuthorizationException;
+
+  /**
+   * Get aprovider for user to perform calls to a resource provided by a given url.
```
Comment (robson-sanchez) [inline integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java]: Get an OAuth1 provider for user to perform calls to an external system resource.

File: integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java [hunk 5]
```diff
@@ -40,4 +44,24 @@
...
+   * @param url Resource url.
```
Comment (robson-sanchez) [inline integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java]: Integration URL

File: integration-authorization/src/main/java/org/symphonyoss/integration/authorization/oauth/v1/OAuth1HttpRequestException.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
+package org.symphonyoss.integration.authorization.oauth.v1;
+
+import org.symphonyoss.integration.exception.IntegrationException;
+
+/**
+ * Created by alexandre-silva-daitan on 14/08/17.
+ */
+public class  OAuth1HttpRequestException extends IntegrationException {
```
Comment (robson-sanchez) [inline integration-authorization/src/main/java/org/symphonyoss/integration/authorization/oauth/v1/OAuth1HttpRequestException.java]: Missing javadoc.

File: integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java [hunk 7]
```diff
@@ -22,7 +25,8 @@
    * @return <code>true</code> if the user is authorized.
    * @throws AuthorizationException Thrown in any case of error.
    */
-  boolean isUserAuthorized(String url, Long userId) throws AuthorizationException;
+  boolean isUserAuthorized(String url, Long userId) throws AuthorizationException,
+      OAuth1HttpRequestException;
```
Comment (robson-sanchez) [inline integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java]: OAuth1HttpRequestException is not required in this method.

**Review by robson-sanchez — CHANGES_REQUESTED** [review]

File: integration-authorization/pom.xml (hunk 1 above)
Comment (ardsilva) [reply integration-authorization/pom.xml]: Done!
b75f02c3c299faa3f620c26bab24d4ac4c81ec8d

File: integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java (hunk 7 above)
Comment (ardsilva) [reply integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java]: Done!
f49a7d338eab109e18827a7fb758d97f6ecb762a

File: integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java (hunk 2 above)
Comment (ardsilva) [reply integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java]: Done!
c47c2377663104233fb2cde9af97862f6d3722a9

File: integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java (hunk 3 above)
Comment (ardsilva) [reply integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java]: Done!
e3646af15be76a3653cad8a733bb9b8571c96786

File: integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java (hunk 4 above)
Comment (ardsilva) [reply integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java]: Done!
324c8840c020e0baf1144d71dde667c714775aa9

File: integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java (hunk 5 above)
Comment (ardsilva) [reply integration-authorization/src/main/java/org/symphonyoss/integration/authorization/AuthorizedIntegration.java]: Done!
1219ead9927726f4ebae05a3d9567f82503ba44e

File: integration-authorization/src/main/java/org/symphonyoss/integration/authorization/oauth/v1/OAuth1HttpRequestException.java (hunk 6 above)
Comment (ardsilva) [reply integration-authorization/src/main/java/org/symphonyoss/integration/authorization/oauth/v1/OAuth1HttpRequestException.java]: Done!
2fb887e521830b7c4bc7b1d93f08b687cc644e39

File: integration-authorization/src/main/java/org/symphonyoss/integration/authorization/oauth/v1/OAuth1HttpRequestException.java [hunk 8]
```diff
@@ -3,6 +3,8 @@
 import org.symphonyoss.integration.exception.IntegrationException;
 
 /**
+ * Exception used to inform runtime exceptions regarding OAuth1HttpRequestException process.
```
Comment (robson-sanchez) [inline integration-authorization/src/main/java/org/symphonyoss/integration/authorization/oauth/v1/OAuth1HttpRequestException.java]: Checked exception to report HTTP error response during the API call to an external system using OAuth1 protocol.

**Review by robson-sanchez — CHANGES_REQUESTED** [review]

File: integration-authorization/src/main/java/org/symphonyoss/integration/authorization/oauth/v1/OAuth1HttpRequestException.java (hunk 8 above)
Comment (ardsilva) [reply integration-authorization/src/main/java/org/symphonyoss/integration/authorization/oauth/v1/OAuth1HttpRequestException.java]: Done!
775e388858f66fe646dbb7e2a011dc23d93be3ae

**Review by robson-sanchez — APPROVED** [review]

**Pull request merged**